vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/model/BlobFilter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.azure.storage.blob.model;
+
+import org.jbehave.core.annotations.AsParameters;
+import org.vividus.bdd.steps.StringComparisonRule;
+
+@AsParameters
+public class BlobFilter
+{
+    private String blobNamePrefix;
+    private StringComparisonRule blobNameFilterRule;
+    private String blobNameFilterValue;
+    private Long resultsLimit;
```

Comment:
`Optional` was actual when I have multiple filters in a table. Like this:
```
When I filter blobs by:
| blobNamePrefix  | blobNameFilterRule | blobNameFilterValue |
| data1/                 |                                  |                                    |
| data2/                 | contains                    | file-key.txt                  |
in container `$containerName` of storage account `$storageAccountKey` and save result to $scopes variable `$variableName`
```
 If I use Optional<blobNamePrefix> - this mean I can use step with empty <blobNamePrefix> filter - not clear for what.
```
When I filter blobs by:
| blobNamePrefix  |
|                             |
in container `$containerName` of storage account `$storageAccountKey` and save result to $scopes variable `$variableName`
```

Reply:
> If I use Optional - this mean I can use step with empty filter - not clear for what.

that's the matter of validation

in general the goal of `Optional` here is to make API clearer, because `Optional` is a clear indication whether field is nullable or not

Reply:
So I guess we don't need nullable filed here. I just either specify needed column or not.

Reply:
All invocations in the step like `Optional.ofNullable(filter.getBlobNamePrefix())` shows that such fields are nullable, it means that they should be `Optional<..>` in usual case, but I haven't checked here whether conversion from `ExamplesTable` to `Optional<..>` field works properly

Reply:
Changed POJO to avoid `Optional.ofNullable`.